File: 백준/Silver/2018. 수들의 합 5/수들의 합 5.java
import java.io.*;
import java.util.*;


public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        
        int n = Integer.parseInt(br.readLine());
        int cnt = 1;
        for (int i = 1; i <= n/2; i++) {
            int e = i + 1;
            int sum = i;
            while (sum <= n) {
                if (sum == n) cnt++;
                sum += e;
                e++;
            }
        }
        
        bw.write(cnt + "\n");
        bw.flush();
        bw.close();
    }
}

File: 백준/Silver/2018. 수들의 합 5/수들의 합 5.py
import sys

input = sys.stdin.readline
print = sys.stdout.write

n = int(input())
cnt = 1
start, end = 1, 1
sum_val = 1
while end < n:
    if (sum_val == n):
        cnt += 1
        end += 1
        sum_val += end
    elif (sum_val < n):
        end += 1
        sum_val += end 
    else:
        sum_val -= start
        start += 1
        
            
                    
            
            
print(str(cnt))